internal/v6provider/resource_load_balancer_test.go

```diff
@@ -85,6 +86,203 @@ func TestAccKatapultLoadBalancer_basic(t *testing.T) {
 					),
 				),
 			},
+			{
+				ResourceName:      "katapult_load_balancer.main",
+				ImportState:       true,
+				ImportStateVerify: false,
+				ImportStateCheck: func(s []*terraform.InstanceState) error {
+					if len(s) != 1 {
+						return fmt.Errorf(
+							"expected 1 instance state, got %d", len(s),
+						)
+					}
+
+					if s[0].ID == "" {
+						return fmt.Errorf("instance state ID is empty")
+					}
+
+					if s[0].Attributes["name"] != name {
+						return fmt.Errorf(
+							"instance state name is %q, expected %q",
+							s[0].Attributes["name"], name,
+						)
+					}
+
+					return nil
+				},
+			},
+		},
+	})
+}
+
+func TestAccKatapultLoadBalancer_basicWithRule(t *testing.T) {
+	tt := newTestTools(t)
+
+	name := tt.ResourceName()
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:                 func() { testAccPreCheck(t) },
+		ProtoV6ProviderFactories: tt.ProviderFactories,
+		CheckDestroy:             testAccCheckKatapultLoadBalancerDestroy(tt),
+		Steps: []resource.TestStep{
+			{
+				Config: undent.Stringf(`
+					resource "katapult_load_balancer" "main" {
+					  name = "%s"
+					  rules = [
+						{
+							destination_port = 8080
+							listen_port = 80
+							protocol = "HTTP"
+							passthrough_ssl = false
+						}
+					  ]
+					}`,
+					name,
+				),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					testAccCheckKatapultLoadBalancerExists(
+						tt, "katapult_load_balancer.main",
+					),
+					resource.TestCheckResourceAttr(
+						"katapult_load_balancer.main", "name", name,
+					),
+					resource.TestCheckResourceAttr(
+						"katapult_load_balancer.main",
+						"resource_type",
+						string(core.VirtualMachinesResourceType),
+					),
+					resource.TestCheckResourceAttr(
+						"katapult_load_balancer.main",
+						"rules.0.destination_port",
+						"8080",
+					),
+					resource.TestCheckResourceAttr(
+						"katapult_load_balancer.main",
+						"rules.0.listen_port",
+						"80",
+					),
+					resource.TestCheckResourceAttr(
+						"katapult_load_balancer.main",
+						"rules.0.protocol",
+						"HTTP",
+					),
+					resource.TestCheckResourceAttr(
+						"katapult_load_balancer.main",
+						"rules.0.passthrough_ssl",
+						"false",
+					),
+					resource.TestCheckResourceAttr(
+						"katapult_load_balancer.main",
+						"rules.0.check_enabled",
+						"false",
+					),
+				),
+			},
+			{
+				ResourceName:      "katapult_load_balancer.main",
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+func TestAccKatapultLoadBalancer_basicWithRules(t *testing.T) {
```

Comment:
We don't have any tests around assigning a LB to VMs, VM Groups, or tags. We should cover at least VMs and VM Groups, but as we don't have a resource or data source for tags yet, we don't have any means of getting a tag ID to test with.